kernel/init.c

```diff
@@ -331,8 +331,8 @@ static char *prepare_multithreading(void)
 	_kernel.ready_q.cache = &z_main_thread;
 #endif
 	stack_ptr = z_setup_new_thread(&z_main_thread, z_main_stack,
-				       CONFIG_MAIN_STACK_SIZE, bg_thread_main,
-				       NULL, NULL, NULL,
+				       CONFIG_MAIN_STACK_SIZE + CONFIG_COVERAGE_STACK_SIZE,
+					   bg_thread_main, NULL, NULL, NULL,
```

Comment:
Nit (style issue): unnecessary indentation change (you seem to be using TAB with the column width of 4; please use 8).